### PR TITLE
Updates to Proof System Docs

### PR DESCRIPTION
**Pull request by pdg744**
This PR started in response to some broken links around receipt types on the proof system overview page. 

While fixing those links, I went ahead and made some revisions to the docs to reflect the three layer proving system. This includes adding some information about our Groth16Receipts, and revising verbiage for our STARK protocol to reflect the fact that it's one piece in the picture. 

This PR also adds the following terms to the terminology page:
  - deterministic builds
  - Groth16
  - Groth16 Receipt
  - Groth16 Circuit
  - Recursion Circuit
  - RISC-V Circuit

Note: There's currently a [ReceiptClaim] link on the proof system page which is broken and which this PR does not address. I think this will work as soon as 0.20 lands; folks should be aware when editing docs that our proof system docs aren't currently part of the versioned docs system 

TODO - images aren't displaying in vercel preview.

### CODE REVIEW

File: website/docs/proof-system/proof-system.md [hunk 1]
```diff
@@ -5,68 +5,71 @@ slug: ./
...
 
-A [Receipt] can take two main forms:
+### Types of Receipts
+A [Receipt] can take four main forms.
 
-- It can be represented by a collection of [SegmentReceipts], each of which proves a single [Segment].
-  Collectively, the [SegmentReceipts] prove the validity of the full [Session].
-- It can also be represented by a single [SuccinctReceipt], proving the validity of the entire session.
-  Using recursive proving, any number of [SegmentReceipts] can be compressed into a single [SuccinctReceipt].
+1. A **[flat receipt]** is a vector of [ZK-STARK]s, one for each [segment]. Segments & segment proofs are constructed by the [RISC-V Circuit].
```
Comment (flaub) [inline website/docs/proof-system/proof-system.md:23]: We renamed this to "Composite" receipt

Comment (pdg744) [reply website/docs/proof-system/proof-system.md:23]: i.e. for 0.20?

Comment (flaub) [reply website/docs/proof-system/proof-system.md:23]: Checking with @nategraf but I thought we did that in `0.19`

Comment (pdg744) [reply website/docs/proof-system/proof-system.md:24]: I wrote this based on what's written [here](https://docs.rs/risc0-zkvm/0.19.1/risc0_zkvm/enum.InnerReceipt.html)

Comment (flaub) [reply website/docs/proof-system/proof-system.md:24]: OK, so the `docs/` directory is considered (unversioned), so I'm not sure how we should deal with name changes like this.

Comment (pdg744) [reply website/docs/proof-system/proof-system.md:24]: I'm not strongly opinionated, but I guess it seems nice to me to maintain versioned history for these proof system docs.

File: website/docs/proof-system/proof-system.md [hunk 2]
```diff
@@ -5,68 +5,78 @@ slug: ./
 
 # Proof System Overview
 
+<center> <img src="assets/receipt.png" width="600"/> </center>
+<br/>
```
Comment (tzerrell) [inline website/docs/proof-system/proof-system.md:9]: Going by how our STARK By Hand docs work, I think we can do this like
```suggestion
![A diagram of a receipt as the output of a zkVM program](assets/receipt.png)
```

**Review by tzerrell — COMMENTED** [review]
Looking good!

File: website/docs/proof-system/proof-system.md (hunk 2 above)
Comment (pdg744) [reply website/docs/proof-system/proof-system.md:9]: Thanks for the suggestion -- the place I'm feeling stuck is when I try to specify a non-default size display...

Comment (tzerrell) [reply website/docs/proof-system/proof-system.md:9]: Resizing sounds possible but cumbersome: https://stackoverflow.com/questions/58265585/change-image-size-in-docusaurus

Comment (pdg744) [reply website/docs/proof-system/proof-system.md:9]: Hmm I took a stab at this and it wasn't immediately obvious. I'm going to go ahead and accept the suggestion you proposed and if the sizing looks bad we can fix it on a future PR

File: website/docs/proof-system/proof-system.md [hunk 3]
```diff
@@ -5,68 +5,77 @@ slug: ./
...
 
-Cryptographically each [SegmentReceipt] or [SuccinctReceipt] is a [ZK-STARK].
-The details of the RISC Zero ZK-STARK are described in our [ZKP Whitepaper] and in this [Sequence Diagram].
+1. A **[flat receipt]** is a vector of [ZK-STARK]s, one for each [segment]. Segments & segment proofs are constructed by the [RISC-V Circuit].
+2. A **[succinct receipt]** is a single [ZK-STARK] proving an entire [Session]. A succinct receipt is formed by aggregating the proofs from a flat receipt. This is accomplished using the [Recursion Circuit]. Users can also aggregate multiple succinct receipts into a single succinct receipt using [proof composition].
+3. A **[Groth16 receipt]** is a single [Groth16] proof for an entire [Session]. A Groth16Receipt is formed by verifying a succinct receipt using RISC Zero's [Groth16 circuit].
+4. A **[fake receipt]** doesn't contain any proof at all. This feature is offered to enable rapid prototyping. See [Dev Mode] for more information.
 
-## Learn More
+<center> <img src="assets/proof-system-layers.png"/> </center>
```
Comment (pdg744) [inline website/docs/proof-system/proof-system.md:28]:
```suggestion
![The layers of the RISC Zero proving system](assets/proof-system-layers.png)
```

**Review by tzerrell — APPROVED** [review]

**Pull request merged**